src/ir_Technibel.cpp

```diff
@@ -0,0 +1,416 @@
+          mode_reversed);
+  setFan(getFan());  // Re-force any fan speed constraints.
+  // Restore previous temp settings for cool mode.
+  setTemp(_saved_temp, _saved_temp_units);
+}
+
+// Convert a standard A/C mode into its native mode.
+uint8_t IRTechnibelAc::convertMode(const stdAc::opmode_t mode) {
+  switch (mode) {
+    case stdAc::opmode_t::kCool:
+      return kTechnibelAcCool;
+    case stdAc::opmode_t::kHeat:
+      return kTechnibelAcHeat;
+    case stdAc::opmode_t::kDry:
+      return kTechnibelAcDry;
+    case stdAc::opmode_t::kFan:
+      return kTechnibelAcFan;
+    default:
+      return kTechnibelAcCool;
+  }
+}
+
+// Convert a native mode to it's common equivalent.
+stdAc::opmode_t IRTechnibelAc::toCommonMode(const uint8_t mode) {
+  switch (mode) {
+    case kTechnibelAcCool:  return stdAc::opmode_t::kCool;
+    case kTechnibelAcHeat:  return stdAc::opmode_t::kHeat;
+    case kTechnibelAcDry:   return stdAc::opmode_t::kDry;
+    case kTechnibelAcFan:   return stdAc::opmode_t::kFan;
+    default:                return stdAc::opmode_t::kAuto;
+  }
+}
+
+void IRTechnibelAc::setSwing(const bool on) {
+  setBit(&remote_state, kTechnibelAcSwingBit, on);
+}
+
+bool IRTechnibelAc::getSwing(void) {
+  return GETBIT64(remote_state, kTechnibelAcSwingBit);
+}
+
+// Convert a standard A/C swing into its native swing.
+bool IRTechnibelAc::convertSwing(const stdAc::swingv_t swing) {
+  switch (swing) {
+    case stdAc::swingv_t::kOff:
+      return false;
+    case stdAc::swingv_t::kAuto:
+      return true;
+    default:
+      return true;
+  }
+}
+
+// Convert a native swing to it's common equivalent.
+stdAc::swingv_t IRTechnibelAc::toCommonSwing(const bool swing) {
+  if (swing)
+    return stdAc::swingv_t::kAuto;
+  else
+    return stdAc::swingv_t::kOff;
+}
+
+void IRTechnibelAc::setSleep(const bool on) {
+  setBit(&remote_state, kTechnibelAcSleepBit, on);
+}
+
+bool IRTechnibelAc::getSleep(void) {
+  return GETBIT64(remote_state, kTechnibelAcSleepBit);
+}
+
+void IRTechnibelAc::setTimerEnabled(const bool on) {
+  setBit(&remote_state, kTechnibelAcTimerEnableBit, on);
+}
+
+bool IRTechnibelAc::getTimerEnabled(void) {
+  return GETBIT64(remote_state, kTechnibelAcTimerEnableBit);
+}
+
+// Set the timer to shutdown the device in nr of minutes.
+// Args:
+//   nr_of_hours: Total nr of hours to wait before shutting down the device.
+//               (Max 24 hrs)
+void IRTechnibelAc::setTimer(const uint8_t nr_of_hours) {
+  uint8_t value = std::min(kTechnibelAcTimerMax, nr_of_hours);
+  uint8_t value_reversed = reverseBits(value, kTechnibelAcHoursSize);
+  setBits(&remote_state, kTechnibelAcTimerHoursOffset, kTechnibelAcHoursSize,
+          value_reversed);
+  // Enable or not?
+  setTimerEnabled(value > 0);
+}
+
+uint8_t IRTechnibelAc::getTimer(void) {
+  uint8_t timer = GETBITS64(remote_state, kTechnibelAcTimerHoursOffset,
+                            kTechnibelAcHoursSize);
+  return reverseBits(timer, kTechnibelAcHoursSize);
```

Comment:
Similarly, please return the `nr_of_mins` instead of hours, to be consistent with other A/C protocols.
i.e.
`return reverseBits(timer, kTechnibelAcHoursSize) * 60;`